server/controllers/roleController.js

```diff
@@ -1,77 +1,99 @@
 import db from '../models';
 
 const RoleController = {
-  createNewRole(req, res) {
-    let roleInfo = {};
-    if (req.body.id) {
-      roleInfo = { title: req.body.title, id: req.body.id };
-    } else { roleInfo = { title: req.body.title }; }
-    db.roles
-      .create(roleInfo)
+  createNewRole(request, response) {
+    let roleData = {};
+    if (request.body.id) {
+      roleData = { title: request.body.title, id: request.body.id };
+    } else { roleData = { title: request.body.title }; }
+    if (!request.body.title) {
+      return response.status(400).send({ message: 'Title cannot be blank' });
+    }
+    const title = request.body.title;
+    db.roles.findOne({
+      where: { title }
+    })
+    .then((result) => {
+      if (result) {
+        return response.status(409).send({
+          success: false,
+          message: 'Role already exists'
+        });
+      }
+      return db.roles
+      .create(roleData)
       .then((role) => {
-        res.status(200).send({
+        response.status(200).send({
           message:
-          'The role was created successfully',
+          'The role was successfully created',
           role
         });
       })
       .catch((err) => {
-        res.status(400).send({ message: 'error', err });
+        response.status(400).send({ message: 'error', err });
       });
+    });
   },
 
-  updateRole(req, res) {
+  updateRole(request, response) {
     db.roles
-      .findById(req.params.id)
+      .findById(request.params.id)
       .then((role) => {
         if (!role) {
-          return res.status(404).send({ message: 'Role was not found' });
+          return response.status(404).send({ message: 'Cannot update a role that does not exist' });
+        }
+        const title = request.body.title;
+        if (!title) {
+          return response.status(404).send({ message: 'You need to write the Title you want to update' });
```

Comment:
Error code of 400 would be more appropriate here seeing as there is isn't an attempt to actually find the title in the DB.